### PR TITLE
Fix sawadm restore

### PR DESCRIPTION
**Pull request by peterschwarz**
Updates to protobuf broke the reading of the length-delimited backup file. This PR fixes the issue.

Additionally, adds context to many of the blockstore commands error messages.

### CODE REVIEW

File: adm/src/commands/blockstore.rs [hunk 1]
```diff
@@ -202,11 +207,13 @@ fn run_show_command<'a>(args: &ArgMatches<'a>) -> Result<(), CliError> {
     }
     .map_err(|err| CliError::ArgumentError(format!("Error getting block: {}", err)))?;
 
-    let block_wrapper = BlockWrapper::try_from(block)
-        .map_err(|err| CliError::EnvironmentError(format!("{}", err)))?;
+    let block_wrapper = BlockWrapper::try_from(block).map_err(|err| {
+        CliError::EnvironmentError(format!("failed to great block wrapper: {}", err))
```
Comment (agunde406) [inline adm/src/commands/blockstore.rs]: Should this say "get block wrapper"

File: adm/src/commands/blockstore.rs [hunk 2]
```diff
@@ -328,42 +359,54 @@ fn run_stats_command<'a>(args: &ArgMatches<'a>) -> Result<(), CliError> {
...
+        })?;
 
         loop {
             for batch in &block.batches {
                 for txn in &batch.transactions {
                     let txn_header: TransactionHeader = protobuf::parse_from_bytes(&txn.header)
-                        .map_err(|err| CliError::ParseError(format!("{}", err)))?;
+                        .map_err(|err| {
+                            CliError::ParseError(format!(
+                                "failed to parse transaction header for trensaction {}: {}",
```
Comment (agunde406) [inline adm/src/commands/blockstore.rs]: trensaction -> transaction

File: adm/src/commands/blockstore.rs (hunk 1 above)
Comment (peterschwarz) [reply adm/src/commands/blockstore.rs]: ha - actually, it should say "create".  Fixed.

File: adm/src/commands/blockstore.rs (hunk 2 above)
Comment (peterschwarz) [reply adm/src/commands/blockstore.rs]: Fixed

**Pull request merged**